### PR TITLE
 TCC-15474 Add Reporting paths and operations

### PR DESCRIPTION
**Pull request by EAddario**

### CODE REVIEW

File: src/main/java/com/currencycloud/client/model/Iban.java [hunk 1]
```diff
@@ -134,20 +140,36 @@ public void setUpdatedAt(Date updatedAt) {
...
-                .appendField("currency", currency)
-                .appendField("accountHolderName", accountHolderName)
-                .appendField("bankInstitutionName", bankInstitutionName)
-                .appendField("bankInstitutionAddress", bankInstitutionAddress)
-                .appendField("bankInstitutionCountry", bankInstitutionCountry)
-                .appendField("bicSwift", bicSwift)
-                .appendField("createdAt", createdAt)
-                .appendField("updatedAt", updatedAt)
-                .toString();
+        ObjectMapper objectMapper = new ObjectMapper();
```
Comment (jonathancouchman) [inline src/main/java/com/currencycloud/client/model/Iban.java]: objectMapper should really be declared "final", also map below

File: src/main/java/com/currencycloud/client/model/Account.java [hunk 2]
```diff
@@ -234,29 +239,38 @@ public void setSettlementType(String settlementType) {
...
-                .appendField("identificationType", identificationType)
-                .appendField("identificationValue", identificationValue)
-                .appendField("shortReference", shortReference)
-                .appendField("apiTrading", apiTrading)
-                .appendField("onlineTrading", onlineTrading)
-                .appendField("phoneTrading", phoneTrading)
-                .appendField("processThirdPartyFunds", processThirdPartyFunds)
-                .appendField("settlementType", settlementType)
-                .toString();
+        ObjectMapper objectMapper = new ObjectMapper();
```
Comment (jonathancouchman) [inline src/main/java/com/currencycloud/client/model/Account.java]: it is more succinct to do

final ObjectMapper objectMapper = new ObjectMapper().disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS).setDateFormat(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX"));

Also date format string should probably be defined as a private static final data member or even as a global static property as it is defined in many classes

File: src/main/java/com/currencycloud/client/CurrencyCloud.java [hunk 3]
```diff
@@ -718,6 +719,7 @@ Ibans findIbans(
     /** Retrieve IBAN of Sub-Account(s) */
     @GET
     @Path("ibans/subaccounts/{id}")
+    @Deprecated
```
Comment (jonathancouchman) [inline src/main/java/com/currencycloud/client/CurrencyCloud.java]: Should the reason/alternative to the deprecation be documented in the javadoc?

File: src/main/java/com/currencycloud/client/model/Iban.java (hunk 1 above)
Comment (EAddario) [reply src/main/java/com/currencycloud/client/model/Iban.java]: Agreed. Will implement as suggested

File: src/main/java/com/currencycloud/client/model/Account.java (hunk 2 above)
Comment (EAddario) [reply src/main/java/com/currencycloud/client/model/Account.java]: Agreed. Will implement as suggested

File: src/main/java/com/currencycloud/client/CurrencyCloud.java (hunk 3 above)
Comment (EAddario) [reply src/main/java/com/currencycloud/client/CurrencyCloud.java]: Reason and alternative are specified in the implementation class (CurrencyCloudClient). Interface (CurrencyCloud) is marked for completeness and consistency

**Pull request merged**